### ISSUE TITLE
LARSOptimizer - Layer-wise decomposition

### ISSUE THREAD

**Issue by Tony-Y** [tensorflow/tensorflow#32125]
LARSOptimizer provides the learning rate control used in LARS, but we need layer-wise decomposed parameters of a neural network. how do we do the layer-wise decomposition for a pre-defined architecture?

LARS: https://arxiv.org/abs/1708.03888 | [Release 1.14 code](https://github.com/tensorflow/tensorflow/blob/release_1.14.0/tensorflow/contrib/opt/python/training/lars_optimizer.py)
LAMB: https://arxiv.org/abs/1904.00962 | [Unofficial code](https://github.com/ymcui/LAMB_Optimizer_TF)
NovoGrad: https://arxiv.org/abs/1905.11286 | [OpenSeq2Seq code](https://github.com/NVIDIA/OpenSeq2Seq/blob/master/open_seq2seq/optimizers/novograd.py)

e.g. [Caffe's LARC](https://docs.nvidia.com/deeplearning/frameworks/caffe-user-guide/index.html#larc)

A possible TensorFlow implementation for conv2d:
```python
import numpy as np
import tensorflow as tf
from tensorflow.python.framework import ops

def lars_conv2d():
    input = tf.placeholder(tf.float32, shape=[16,224,224,3])
    with tf.name_scope('conv1_1') as scope:
        kernel_size = 7*7*3*32
        bias_size = 32
        total_size = kernel_size + bias_size
        params = tf.Variable(tf.constant(0.0, shape=[total_size], dtype=tf.float32),
                             trainable=True, name='weights_and_biases')
        kernel = tf.reshape(
                    tf.slice(params, [0], [kernel_size], name='weights'),
                    [7, 7, 3, 32])
        conv = tf.nn.conv2d(input, kernel, [1, 1, 1, 1], padding='SAME')
        biases = tf.slice(params, [kernel_size], [bias_size], name='biases')
        bias = tf.nn.bias_add(conv, biases)
        conv1 = tf.nn.relu(bias, name=scope)

    init_op = tf.variables_initializer(var_list=[params])
    with tf.Session() as sess:
        sess.run(init_op)
        print(np.sum(sess.run(conv1, {input: np.zeros([16,224,224,3])})))

lars_conv2d()
```

**Comment by jvishnuvardhan** [comment]
This is not Build/Installation or Bug/Performance issue. Please post this kind of support questions at [Stackoverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a big community to support and learn from your questions. GitHub is mainly for addressing bugs in installation and performance. Thanks!

**Comment by Tony-Y** [comment]
https://stackoverflow.com/questions/57781188/larsoptimizer-layer-wise-decomposition

I am waiting answers from the TensorFlow community.

**Comment by zhenzhen2017** [comment]
![image](https://user-images.githubusercontent.com/33881637/87132980-3f092380-c2c9-11ea-9d75-2c38ffa70a31.png)
maybe we can fix this bug according to reference:
https://people.eecs.berkeley.edu/~youyang/lars_optimizer.py
